pkg/services/baremetal/host/host.go

```diff
@@ -588,6 +588,31 @@ func (s *Service) actionRegistering() actionResult {
 		return s.recordActionFailure(infrav1.RegistrationError, err.Error())
 	}
 
+	// Check RAID for the second time.
+	// This happens for new machines which don't have rootDeviceHints.
+	// See "tworaidchecks" for the other place.
+	msg = ""
+	if s.scope.HetznerBareMetalHost.Spec.Status.InstallImage == nil {
+		msg = "Internal Error: HetznerBareMetalHost.Spec.Status.InstallImage is nil. setHostSpec() not called yet?"
+	} else if s.scope.HetznerBareMetalHost.Spec.Status.InstallImage.Swraid != 0 &&
+		len(s.scope.HetznerBareMetalHost.Spec.RootDeviceHints.Raid.WWN) < 2 {
+		msg = "HetznerBareMetalHost.Spec.Status.InstallImage.Swraid is active. List spec.rootDevideHints.raid.wwn must contain at least two entries."
+	} else if s.scope.HetznerBareMetalHost.Spec.Status.InstallImage.Swraid == 0 &&
+		len(s.scope.HetznerBareMetalHost.Spec.RootDeviceHints.Raid.WWN) > 0 {
+		msg = "HetznerBareMetalHost.Spec.Status.InstallImage.Swraid is not active. List spec.rootDevideHints.raid.wwn must be empty."
+	}
+	if msg != "" {
+		s.scope.Logger.Info(msg)
+		conditions.MarkFalse(
+			s.scope.HetznerBareMetalHost,
+			infrav1.RootDeviceHintsValidatedCondition,
+			infrav1.StorageDeviceNotFoundReason,
```

Comment:
this is not the appropriate reason I guess

Reply:
I use `infrav1.ValidationFailedReason,` now